examples/encoding/plot_custom_encoder.py

```diff
@@ -0,0 +1,179 @@
+sig_event = modulate_noise(stim_events[:, 0], sfreq, delays_sig,
+                           weights=weights, snr=snr)
+sig_event = mne.io.RawArray(sig_event[np.newaxis, :], info)
+
+# Combine the two together
+sig_combined = mne.io.RawArray(sig_event._data + sig_cont._data, info)
+stim_combined = np.hstack([stim_events, stim_cont])
+
+# -- Preparing model features --
+# Iterate through our events-based and continuous datasets
+data_iterator = [(stim_events, sig_event, 'Events'),
+                 (stim_cont, sig_cont, 'Continuous'),
+                 (stim_combined, sig_combined, 'Combined')]
+
+# Define a training / test set of indices
+ixs = np.arange(sig_cont.n_times)
+ix_split = int(.8 * ixs.shape[0])
+tr = ixs[:ix_split]
+tt = ixs[ix_split:]
+
+
+# -- Define preprocessing pipelines for X/y --
+# For creating time lags
+delays_model = np.arange(.2, -.4, -.01)
+delayer = DataDelayer(delays=delays_model, sfreq=sig_cont.info['sfreq'])
+
+# To subset training data
+sub_tr = DataSubsetter(tr)
+sub_tt = DataSubsetter(tt)
+
+# Now put them together into pipelines
+pipe_x = Pipeline([('delays', delayer), ('subset', sub_tr)])
+pipe_y = Pipeline([('subset', sub_tr)])
+preproc_x_pred = Pipeline([('delays', delayer), ('subset', sub_tt)])
+
+
+# -- Fit the model, iterating through alphas --
+alphas = np.logspace(0, 2, 4)
+_, axs_coef = plt.subplots(3, len(alphas), figsize=(4 * len(alphas), 8),
+                           sharex=True, sharey=True)
+_, axs_pred = plt.subplots(3, len(alphas), figsize=(4 * len(alphas), 8),
+                           sharex=True, sharey=False)
+for alpha, axcol_coef, axcol_pred in zip(alphas, axs_coef.T, axs_pred.T):
+    axcol_coef[0].set_title('Alpha: %s' % np.round(np.log10(alpha), 2))
+    axcol_pred[0].set_title('Alpha: %s' % np.round(np.log10(alpha), 2))
+
+    # Define our estimator and create the model
+    clf = Ridge(alpha=alpha)
+    mod = EncodingModel(est=clf, preproc_x=pipe_x, preproc_y=pipe_y)
+
+    # Fit the model and plot coefficients
+    iter_fit = zip(data_iterator, axcol_coef, axcol_pred)
+    for (X, y, title), ax_coef, ax_pred in iter_fit:
+        mod.fit(X, y._data.T)
+
+        # Plot coefficients
+        ax = ax_coef
+        coefs = mod.est._final_estimator.coef_.squeeze()
+
+        if title != 'Combined':
+            ax.plot(delays_model, coefs)
+            ax.plot(delays_sig, weights)
+            ax.set_xlabel('Lag (s)')
+        else:
+            for i, i_weight in enumerate(coefs.reshape([2, -1])):
+                # Plot both sets of weights in this case
+                ax.plot(delays_model, i_weight + (i * 10))
+                ax.plot(delays_sig, weights + (i * 10))
+
+        # Plot predictions (on same data, so kinda cheating)
+        predicted = mod.predict(X, preproc_x=preproc_x_pred)
+        ax_pred.plot(y._data[:, tt].squeeze(), color='k', alpha=.3)
+        ax_pred.plot(predicted.squeeze(), color='r')
+
```

Comment:
I'm adding an example in the next commit that does continuous regression using the MEG dataset. Basically it picks one channel as the output, and uses all other channels as input. It's a lot simpler than the current "custom_encoder" example. Here are the plots it makes...LMKWYT:

![image](https://cloud.githubusercontent.com/assets/1839645/16364903/9d680da0-3ba6-11e6-8bf3-cdab3f5c34c9.png)

![image](https://cloud.githubusercontent.com/assets/1839645/16364892/7ee772c6-3ba6-11e6-8b81-cce3b74a995d.png)